src/Pact/Eval.hs

```diff
@@ -886,11 +886,35 @@ applyPact i app (TList steps _ _) PactStep {..} = do
 
   writeRow i Write Pacts _psPactId $ if done then Nothing else Just resultState
 
+  unlessExecutionFlagSet FlagDisablePact40 $ emitXChainEvents _psResume resultState
+
   return result
 
 applyPact _ _ t _ = evalError' t "applyPact: invalid defpact body, expected list of steps"
 
 
+-- | Synthesize events for cross chain.
+emitXChainEvents :: Maybe Yield -> PactExec -> Eval e ()
+emitXChainEvents mResume PactExec {..} = do
```

Comment:
Is it acceptable that `mResume` is `Nothing` while `_peYield` yields (pardon the obvious pun) a `Just` value?

Reply:
Yes, in fact that's expected. I'll update comments